controllers/util/provider.go

```diff
@@ -19,14 +19,15 @@ import (
 
 // GenerateOnboardingToken generates a token valid for a duration of "tokenLifetimeInHours".
 // The token content is predefined and signed by the private key which'll be read from supplied "privateKeyPath".
-func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string) (string, error) {
+func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath, quota string) (string, error) {
```

Comment:
Quota shouldn't be a string, it should be wither a number (assuming a specific unit) or a size object.
Also it should preferably be optional (a pointer) so we can generate a token that does not have a quota

Reply:
I have considered this as a string because in ClusterResourceQuota we also need to mention the unit. It will be better if we get that from the UI itself.

Reply:
Even it that case this should not be a string but a k8s quantity object

Reply:
Okay will do the changes according